packages/worker/src/isolate-builder.ts

```diff
@@ -0,0 +1,151 @@
+  }
+
+  protected async genEntrypoint(target: string, workflows: string[]): Promise<void> {
+    let code = dedent`
+      const init = require('@temporalio/workflow/lib/init');
+      const internals = require('@temporalio/workflow/lib/internals');
+      internals.state.activityDefaults = ${JSON.stringify(this.activityDefaults)};
+
+      const workflows = {};
+      export { init, internals, workflows };
+    `;
+    for (const wf of workflows) {
+      code += `workflows[${JSON.stringify(wf)}] = require(${JSON.stringify(path.join(this.workflowsPath, wf))});\n`;
+    }
+    await fs.writeFile(target, code);
+  }
+
+  protected async bundle(entry: string, distDir: string): Promise<void> {
+    await new Promise<void>((resolve, reject) =>
+      webpack(
+        {
+          resolve: {
+            modules: [this.nodeModulesPath],
+            extensions: ['.js'],
+            alias: Object.fromEntries(
+              [...this.activities.keys()].map((spec) => [spec, path.resolve(distDir, 'prebuild', spec)])
+            ),
+          },
+          entry: [entry],
+          mode: 'development',
+          output: {
+            path: distDir,
+            filename: 'main.js',
+            library: 'lib',
+          },
+        },
+        (err, stats) => {
+          const lines = stats.toString({ chunks: false, colors: true }).split('\n');
+          for (const line of lines) {
+            this.logger.info(line);
+          }
+          if (err) {
+            reject(err);
+          } else if (stats.hasErrors()) {
+            reject(new Error('Webpack stats has errors'));
+          } else {
+            resolve();
+          }
+        }
+      )
+    );
+  }
+
+  public async registerActivities(prebuildDir: string): Promise<void> {
+    validateActivityOptions(this.activityDefaults);
+    const serializedOptions = JSON.stringify(this.activityDefaults);
+    for (const [specifier, module] of this.activities.entries()) {
+      let code = dedent`
+        import { scheduleActivity } from '@temporalio/workflow';
+      `;
+      for (const [k, v] of Object.entries(module)) {
+        if (v instanceof Function) {
+          // Activities are identified by their module specifier and name.
+          // We double stringify below to generate a string containing a JSON array.
+          const type = JSON.stringify(JSON.stringify([specifier, k]));
+          // TODO: Validate k against pattern
+          code += dedent`
+            export function ${k}(...args) {
+              return scheduleActivity(${type}, args, ${serializedOptions});
+            }
+            ${k}.type = ${type};
+            ${k}.options = ${serializedOptions};
+          `;
+        }
+      }
+      const modulePath = path.resolve(prebuildDir, `${specifier}.js`);
+      try {
+        await fs.mkdir(path.dirname(modulePath), { recursive: true });
+      } catch (err) {
+        if (err.code !== 'EEXIST') throw err;
```

Comment:
This might be platform sensitive. Alternatively, is there a constant that can be imported for this code? I can't even find docs about how the error is supposed to work...https://nodejs.org/api/fs.html#fs_fspromises_mkdir_path_options